src/cmd-kola

```diff
@@ -46,14 +46,22 @@ if os.path.isfile(old_denylist_path):
     print(f"⛔ Detected an old filename for denylist; please update your filename to `kola-denylist.yaml`")
     denylist_path = old_denylist_path
 
+current_branch = subprocess.check_output(["git", "-C", "src/config", "branch",
+                                          "--show-current"],
+                                         encoding='utf-8').strip()
 if os.path.isfile(denylist_path):
     with open(denylist_path) as f:
         denylist = yaml.safe_load(f)
         for obj in (denylist or []):
-            # if there are any arches specified, skip tests only for those arches. if not skip unconditionally.
-            if basearch in obj.get('arches', [basearch]):
+            if 'branches' in obj and current_branch in obj['branches']:
                 print(f"⚠️  Skipping kola test pattern \"{obj['pattern']}\":")
                 print(f"⚠️  {obj['tracker']}")
+                print(f"⚠️  Test excluded for branch {current_branch}")
+                denylist_args += ['--denylist-test', obj['pattern']]
+            elif 'arches' in obj and basearch in obj['arches']:
+                print(f"⚠️  Skipping kola test pattern \"{obj['pattern']}\":")
+                print(f"⚠️  {obj['tracker']}")
+                print(f"⚠️  Test excluded for basearch {current_branch}")
                 denylist_args += ['--denylist-test', obj['pattern']]
```

Comment:
This seems to have the effect that no tests get denied.I think the earlier check of `if basearch in obj.get('arches', [basearch]):` would always pass so we would drop into `print(f"⚠️  Skipping kola test pattern \"{obj['pattern']}\":")`.